state-synchronizer/src/chunk_response.rs

```diff
@@ -20,8 +20,11 @@ pub enum ResponseLedgerInfo {
     ProgressiveLedgerInfo {
         // LedgerInfo that the corresponding GetChunkResponse is built relative to.
         target_li: LedgerInfoWithSignatures,
-        // LedgerInfo that is the highest LI of the responder
-        highest_li: LedgerInfoWithSignatures,
+        // LedgerInfo that is the highest LI available on the responder
+        // for the epoch of `target_li`
+        // If the responder is in a later epoch compared to the request's `target_li`, `target_li` will be an end-of-epoch LI,
+        // and `highest_li` will be empty
+        highest_li: Option<LedgerInfoWithSignatures>,
```

Comment:
why not make this the end-of-epoch LI instead of None?

Reply:
if the corresponding request for this response is for a past epoch, the `target_li` will already be the end-of-epoch LI, so it would be redundant to specify it here. nothing wrong about repeating that end-of-epoch LI here, just wanted to make this message compact

Reply:
well it's already possible for the target_li to be the same as the highest_li if that's the highest in remote? I don't like special case, I'd expect it's either both cases the highest_li is None or redundant